### PR TITLE
Chain operation could be simplified

### PR DESCRIPTION
**Pull request by remyleone**
#### Brief description of what is fixed or changed

This simplifies boolean chain operation

### CODE REVIEW

**Comment by smichr** [comment]
If all args are comparable numbers this will work, otherwise this is a "gotcha" discussed in the tutorial (search for "chained" [here](http://docs.sympy.org/latest/modules/core.html)).

File: sympy/sets/tests/test_sets.py [hunk 1]
```diff
@@ -632,7 +632,7 @@ def test_finite_basic():
     assert (A <= B) is False
     assert AorB > A and AorB > B
     assert AorB >= A and AorB >= B
-    assert A >= A and A <= A
+    assert A >= A >= A
```
Comment (asmeurer) [inline sympy/sets/tests/test_sets.py]: This test is explicitly testing <=. It should not be changed.

File: sympy/core/tests/test_numbers.py [hunk 2]
```diff
@@ -849,7 +849,7 @@ def test_Infinity_inequations():
     x = Symbol('x')
     b = Symbol('b', finite=True, real=True)
     assert (x < oo) == Lt(x, oo)  # issue 7775
-    assert b < oo and b > -oo and b <= oo and b >= -oo
+    assert oo > b > -oo and b <= oo and b >= -oo
```
Comment (asmeurer) [inline sympy/core/tests/test_numbers.py]: Same here, these tests may be explicitly testing the inequality as written. I would be more cautious about changing the tests, especially tests for inequalities themselves.

**Comment by Abdullahjavednesar** [comment]
@smichr can we merge this now?

File: sympy/sets/tests/test_sets.py (hunk 1 above)
Comment (jksuom) [reply sympy/sets/tests/test_sets.py]: This test should be left as it is.

**Comment by asmeurer** [comment]
@Abdullahjavednesar I didn't see all my comments addressed.

**Pull request closed without merging**